src/entities/SubscriptionSet.ts

```diff
@@ -31,20 +31,16 @@ export class SubscriptionSet extends SubscribeCapable {
     this.options = subscriptionOptions;
     this.eventEmitter = eventEmitter;
     this.pubnub = pubnub;
-    channels
-      .filter((c) => !c.endsWith('-pnpres'))
-      .forEach((c) => {
-        const subscription = this.pubnub.channel(c).subscription(this.options);
-        this.channelNames = [...this.channelNames, ...subscription.channels];
-        this.subscriptionList.push(subscription);
-      });
-    channelGroups
-      .filter((cg) => !cg.endsWith('-pnpres'))
-      .forEach((cg) => {
-        const subscription = this.pubnub.channelGroup(cg).subscription(this.options);
-        this.groupNames = [...this.groupNames, ...subscription.channelGroups];
-        this.subscriptionList.push(subscription);
-      });
+    channels.forEach((c) => {
+      const subscription = this.pubnub.channel(c).subscription(this.options);
+      this.channelNames = [...this.channelNames, ...subscription.channels];
+      this.subscriptionList.push(subscription);
+    });
+    channelGroups.forEach((cg) => {
+      const subscription = this.pubnub.channelGroup(cg).subscription(this.options);
+      this.groupNames = [...this.groupNames, ...subscription.channelGroups];
+      this.subscriptionList.push(subscription);
+    });
```

Comment:
Wouldn't this cause trouble? `Subscription` object will give regular and `-pnpres` channels if it will be required by options.
I would expect that the required set of channels / groups (with presence channels) will be stored in `channelNames` and `groupName` when the filter will be back. Or it didn't work as expected with `filter`?

Reply:
Right now channelNames, groupNames have all required channels and groups.

When filter will be back then -pnpres channels will be filtered out and the same backward compatibility issue will arise.

It will be problem if User provides `-pnpres` channel names along with `receivePresenceEvents: true`
And so we do not recommend it.

Reply:
> On the other hand, sometime it is the only way to subscribe only to presence events.


Need to figure out the use case for it.